src/fparser/two/C99Preprocessor.py

```diff
@@ -0,0 +1,561 @@
+#     # the specification
+#     subclass_names = Program_2003.subclass_names[:]
+#     subclass_names.append('Cpp_If_Construct')
+
+
+# class Cpp_If_Construct(BlockBase):  # 6.10.1 Conditional inclusion
+#     '''
+#     C99 6.10.1 Conditional inclusion
+#
+#     <cpp-if-construct> = <cpp-if-stmt>
+#                             [ <block> ]
+#                          [ <cpp-elif-stmt>
+#                             [ <block> ]
+#                          ]...
+#                          [ <cpp-else-stmt>
+#                             [ <block> ]
+#                          ]
+#                          <cpp-endif-stmt>
+#     '''
+#
+#     subclass_names = []
+#     use_names = ['Cpp_If_Stmt', 'Block', 'Cpp_Elif_Stmt', 'Cpp_Else']
+#
+#     @staticmethod
+#     def match(string):
+#         return BlockBase.match(
+#             Cpp_If_Stmt, [Execution_Part_Construct,
+#                           Cpp_Elif_Stmt,
+#                           Execution_Part_Construct,
+#                           Cpp_Else_Stmt,
+#                           Execution_Part_Construct],
+#             Cpp_Endif_Stmt, string,
+#             enable_cpp_if_construct_hook=True)
+#
+#     def tofortran(self, tab='', isfix=None):
+#         tmp = []
+#         start = self.content[0]
+#         end = self.content[-1]
+#         tmp.append(start.tofortran(tab='', isfix=isfix))
+#         for item in self.content[1:-1]:
+#             if isinstance(item, (Cpp_Elif_Stmt, Cpp_Else_Stmt)):
+#                 tmp.append(item.tofortran(tab='', isfix=isfix))
+#             else:
+#                 tmp.append(item.tofortran(tab=tab+'  ', isfix=isfix))
+#         tmp.append(end.tofortran(tab='', isfix=isfix))
+#         return '\n'.join(tmp)
+
+
+class Cpp_If_Stmt(Base):
+    '''
+    C99 6.10.1 Conditional inclusion
+
+    if-stmt is  # if constant-expression new-line
+                | ifdef identifier new-line
+                | ifndef identifier new-line
+
+    '''
+
+    subclass_names = []
+    use_names = ['Cpp_Macro_Identifier']
+
+    _regex = re.compile(r"#\s*(ifdef|ifndef|if)\b")  # 'if' last, order matters
+
+    @staticmethod
+    def match(string):
+        if not string:
+            return None
+        line = string.strip()
+        found = Cpp_If_Stmt._regex.match(line)
+        if not found:
+            return None
+        kind = found.group()[1:].strip()
+        rhs = line[found.end():].strip()
+        if rhs is None or len(rhs) == 0:
+            return None
+        if kind in ['ifdef', 'ifndef']:
+            rhs = Cpp_Macro_Identifier(rhs)
+        return (kind, rhs)
+
+    def tostr(self):
+        return ('#{} {}'.format(self.items[0], self.items[1]))
+
+
+class Cpp_Elif_Stmt(Base):
+    '''
+    C99 6.10.1 Conditional inclusion
+
+    elif-stmt is  # elif constant-expression new-line
+
+    '''
+
+    subclass_names = []
+
+    _regex = re.compile(r"#\s*elif\b")
+
+    @staticmethod
+    def match(string):
+        if not string:
```

Comment:
Again, docstring describing the method and a description of the arguments please. I'll not add this comment to any subsequent functions/classes with missing docstrings but please add them there too.